Tests/Kernel/TestProcFSWrite.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021, Ben Wiederhake <BenWiederhake.GitHub@gmx.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+#include <fcntl.h>
+#include <sys/prctl.h>
+#include <unistd.h>
+
+TEST_CASE(check_root)
+{
+    auto uid = geteuid();
+    // This test only makes sense as root.
+    EXPECT_EQ(uid, 0u);
+    // If running as setuid, the process is automatically marked as dumpable, which bars access to /proc/self/.
+    // However, that is the easiest guess for a /proc/$PID/ directory, so we'd like to use that.
+    // In order to do so, we need to mark this process as dumpable:
+    EXPECT_EQ(prctl(PR_SET_DUMPABLE, 1, 0), 0);
```

Comment:
Er so. Why do we need to access /proc/self instead of using `String::formatted("/proc/{}/unveil", getpid()):` ?

Reply:
Or is access to /proc/pid-that-resolves-to-seld also blocked by the dumpable flag?

Reply:
Because that makes no difference.

ProcFS says "Oh, someone is trying to access the files of process 42. Is 42 dumpable? No? Oh golly! RAISE THE ALARMS!"

Reply:
Hmm. Since your original goal was to find SystemServer maybe you could walk until /proc/pid/exe shows it's /bin/SystemServer?

Reply:
I also added a `setuid(0)` to make it less likely that dumpability opens a vulnerability.

Reply:
> Hmm. Since your original goal was to find SystemServer maybe you could walk until /proc/pid/exe shows it's /bin/SystemServer?

Sounds like more effort than just doing `setuid(0); prctl(…);`

Also, SystemServer wasn't really a "goal", it just was the easiest way to get *any* process that is guaranteed to exist. This doesn't work in CI, so let's use the test program itself instead.